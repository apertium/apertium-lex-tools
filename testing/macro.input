^hi<x>/ho<x>$ ^yi<n><sg>/ya<n><sg>$ ^ho<y>/ha<y>$ ^yo<n>/blah<q>/bloop<r>$
^a<x>/a<x>$ ^b<y>/b<z>/noodle<z>$
^octopi<x>/a<x>$ ^potato<y>/b<z>/noodle<z>$
