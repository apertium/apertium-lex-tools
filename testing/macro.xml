<lrx>
	<def-macros>
		<def-macro n="n" nodes="2" npar="0">
			<rule>
				<param n="1"/>
				<match tags="n.*"/>
				<param n="2"/>
			</rule>
		</def-macro>
		<def-macro n="s" nodes="0" npar="2">
			<rule>
				<match plemma="1"/>
				<match plemma="2">
					<select lemma="noodle"/>
				</match>
			</rule>
		</def-macro>
		<def-macro n="nest" npar="1">
			<macro n="s">
				<with-param pv="1"/>
				<with-param v="potato"/>
			</macro>
		</def-macro>
	</def-macros>
	<rules>
		<macro n="n">
			<with-param>
				<match lemma="hi"/>
			</with-param>
			<with-param>
				<match lemma="ho"/>
				<match lemma="yo">
					<select lemma="blah"/>
				</match>
			</with-param>
		</macro>
		<macro n="s">
			<with-param v="a"/>
			<with-param v="b"/>
		</macro>
		<macro n="nest">
			<with-param v="octopi"/>
		</macro>
	</rules>
</lrx>
