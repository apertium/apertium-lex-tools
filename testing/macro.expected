^hi<x>/ho<x>$ ^yi<n><sg>/ya<n><sg>$ ^ho<y>/ha<y>$ ^yo<n>/blah<q>$
^a<x>/a<x>$ ^b<y>/noodle<z>$
^octopi<x>/a<x>$ ^potato<y>/noodle<z>$
